src/wireclient.js

```diff
@@ -490,8 +490,81 @@ WireClient.prototype = {
 WireClient.isArrayBufferView = isArrayBufferView;
 
 // Shared request function used by WireClient, GoogleDrive and Dropbox.
-// TODO: Should we use fetch ?
 WireClient.request = function (method, url, options) {
+  if (typeof fetch === 'function') {
+    return WireClient._fetchRequest(method, url, options);
+  } else if (typeof XMLHttpRequest === 'function') {
+    return WireClient._xhrRequest(method, url, options);
+  } else {
+    log('[WireClient] no support for HTTP requests');
+    return Promise.reject('[WireClient] no support for HTTP requests');
+  }
+};
+
+/** options includes body, headers and responseType */
+WireClient._fetchRequest = function (method, url, options) {
+  var syntheticXhr;
+  var responseHeaders = {};
+  var abortController;
+  if (typeof AbortController === 'function') {
+    abortController = new AbortController();
+  }
+  var networkPromise = fetch(url, {
+    method: method,
+    headers: options.headers,
+    body: options.body,
+    signal: abortController ? abortController.signal : undefined
+  }).then(function (response) {
+    log('[WireClient fetch]', response);
+
+    response.headers.forEach(function (value, headerName) {
+      responseHeaders[headerName.toUpperCase()] = value;
+    });
+
+    syntheticXhr = {
+      readyState: 4,
+      status: response.status,
+      statusText: response.statusText,
+      response: undefined,
+      getResponseHeader: function (headerName) {
+        return responseHeaders[headerName.toUpperCase()] || null;
+      },
+      // responseText: 'foo',
+      responseType: options.responseType,
+      responseURL: url,
+    };
+    switch (options.responseType) {
+      case 'arraybuffer':
+        return response.arrayBuffer();
+      case 'blob':
+        return response.blob();
+      case 'json':
+        return response.json();
+      case '':
+      case 'text':
+        return response.text();
+      default:   // document
+        throw new Error("responseType 'document' is not currently supported using fetch");
+    }
+  }).then(function (processedBody) {
+    syntheticXhr.response = processedBody;
+    return syntheticXhr;
+  });
+
+  var timeoutPromise = new Promise(function (resolve, reject) {
+    setTimeout(function () {
+      log('[WireClient] request timed out', url);
```

Comment:
As this promise is raced, but nothing is cancelled, this code will still be run. I'm not sure about potential side effects (for example I had rs-backup not exit the program for me after syncing all items with this branch, but not sure if related). However, the line is always logged in the debug logs, which is definitely not a good idea:

![screenshot from 2018-11-17 11-16-24](https://user-images.githubusercontent.com/842/48660720-87256380-ea5e-11e8-9f11-df367632f45f.png)

Reply:
Good catch!  That log() line should be removed.

Unfortunately, I think I rebased at the wrong time before pushing the commit with this. I'll make a new branch and PR with just my changes.